service/routes.py

```diff
@@ -172,11 +172,34 @@ def get_customer_addresses(customer_id):
 def list_customers():
     """Returns all of the customers"""
     app.logger.info("Request for customer list")
-    category = request.args.get("category")
-    name = request.args.get("first_name")
-
-    results = [customer.serialize() for customer in Customer.all()]
+    
+    all_query_key = ["username", "first_name", "last_name", "addresses"]
+    for key in request.args.keys():
+      if key not in all_query_key:
+        message = {
+            "error": "Conflict",
+            "message": "The query key: '" + key + "' is not supported."
+            }
 
+        return make_response(
+            jsonify(message), status.HTTP_409_CONFLICT
```

Comment:
I'm wondering if this should be a 400, but have not found a solid source indicating what to do in this case yet

Reply:
Fixed.